### PR TITLE
Add pivoting/move documentation

### PR DESCRIPTION
**Pull request by kashifest**
This PR adds the pivoting/move documentation.

### CODE REVIEW

File: docs/pivoting.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# Pivoting/Move
```
Comment (dhellmann) [inline docs/pivoting.md:1]: I'm not sure it's necessary to use both terms every time we refer to the process. Which term is preferred? Let's use that in the title, and mention the alternative once in the introduction, then use the preferred term everywhere else.

File: docs/pivoting.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+* **BMH objects have correct status annotation.**
+
+  Since BMH _status_ holds important information regarding the BMH itself, we
+  have to move BMH with status and reconstruct the BMH with correct status in
+  target  cluster before it is being reconciled. This is now done through BMH
+  status annotation in BMO.
+
+* **Maintain connectivity towards provisioning network.**
+
+  Baremetal machines boot over a network with a dhcp server. This requires
```
Comment (dhellmann) [inline docs/pivoting.md:45]:
```suggestion
  Baremetal machines boot over a network with a DHCP server. This requires
```

File: docs/pivoting.md (hunk 1 above)
Comment (kashifest) [reply docs/pivoting.md]: Agree. I have changed it accordingly.

File: docs/pivoting.md (hunk 2 above)
Comment (kashifest) [reply docs/pivoting.md]: Done

File: docs/pivoting.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
+# Move
```
Comment (dhellmann) [inline docs/pivoting.md:1]: The file is called "pivoting". Can we make the file and title match?

File: docs/pivoting.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+## Prerequisite
+
+* **`clusterctl` is used to initialize both the bootstrap and target cluster.**
+
+  If the provider components are not installed using `clusterctl`, it will not
+  be able to identify the objects to move.  Initializing the cluster using
+  `clusterctl` essentially adds the following labels in the CRDs of each object
+  related.
+
+  ```bash
```
Comment (dhellmann) [inline docs/pivoting.md:27]: this is YAML text, right?

File: docs/pivoting.md (hunk 3 above)
Comment (kashifest) [reply docs/pivoting.md]: Done

File: docs/pivoting.md (hunk 4 above)
Comment (kashifest) [reply docs/pivoting.md]: Indeed. I have changed it.

**Pull request merged**